### ISSUE TITLE
Known-variance masked observations still affect SVD initialization

### ISSUE THREAD

**Issue by Derekamethy** [william-denault/cebmf_torch#76]
## Summary

When `S` is supplied, non-finite or non-positive entries are folded into `self.mask` and the corresponding `Y0` entries are zeroed, so those observations are treated as missing by the likelihood. However, SVD factor initialization still receives the original `self.Y`, and `_impute_nan()` only treats literal `NaN` values as missing.

As a result, changing the value of an observation that has already been excluded by the `S` mask can substantially change the SVD initialization.

Verified on current `main` at `2f00abd36f0b8ec9737c97c51e29cba09028ac0e`.

## Relevant code path

`_setup_known_variance()` folds invalid `S` entries into the effective missing-data mask:

```python
bad = ~torch.isfinite(S) | (S <= 0)
good_f = (~bad).to(self.mask.dtype)
self.mask = self.mask * good_f
self.Y0 = self.Y0 * self.mask
```

But `initialise_factors()` passes the original data tensor to the selected strategy:

```python
initialise_fn = INIT_STRATEGIES[method]
self.L, self.F = initialise_fn(self.Y, self.N, self.P, self.model.K, self.device)
```
and `svd_initialise()` only imputes literal NaNs:

```python
Y_for_init = _impute_nan(Y)
U, S, Vh = torch.linalg.svd(Y_for_init, full_matrices=False)
```

## Minimal reproduction

```python
import warnings
import torch
from cebmf_torch import cEBMF

base = torch.tensor([
    [0.0, 2.0, 3.0],
    [4.0, 5.0, 6.0],
    [7.0, 8.0, 9.0],
], dtype=torch.float32)

Y_pos = base.clone(); Y_pos[0, 0] = 1000.0
Y_neg = base.clone(); Y_neg[0, 0] = -1000.0
S = torch.ones_like(base); S[0, 0] = 0.0

with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    m_pos = cEBMF(Y_pos, K=1, S=S, device=torch.device("cpu"))
    m_neg = cEBMF(Y_neg, K=1, S=S, device=torch.device("cpu"))
print(torch.equal(m_pos.mask, m_neg.mask))  # True
print(torch.equal(m_pos.Y0, m_neg.Y0))      # True
print(m_pos.mask[0, 0].item())              # 0.0

m_pos.initialise_factors("svd")
m_neg.initialise_factors("svd")

R_pos = m_pos.L @ m_pos.F.T
R_neg = m_neg.L @ m_neg.F.T
print((R_pos - R_neg).abs().max().item())
```

Current output:

```text
True
True
0.0
2000.000244140625
```

So the two models have the same effective likelihood data (`mask` and `Y0` are identical), but the value at an observation already marked missing by `S` still dominates the rank-1 SVD initialization.

## Expected behavior

If an observation is documented and handled as missing because its `S` value is invalid, I would expect its raw `Y` value not to influence SVD initialization either.
A useful regression invariant would be: two inputs that differ only at positions where the effective mask is zero should produce the same SVD initialization *reconstruction* (`L @ F.T`), avoiding dependence on arbitrary SVD sign conventions.

## Possible direction

One minimal approach would be to make SVD initialization use the effective missing-data mask as well, for example by presenting masked entries as `NaN` before the existing column-mean imputation step. A more explicitly mask-aware initializer may also be preferable.

I have not opened a PR for this because the desired missing-data initialization policy is a design choice. Would you prefer the existing column-mean SVD initialization to use the effective mask, or a different masked-initialization strategy?
